x-pack/legacy/plugins/siem/public/components/ml/conditional_links/replace_kql_query_location_for_host_page.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RisonValue, encode } from 'rison-node';
+import { decodeRison, isRisonObject } from './rison_helpers';
+import { CONSTANTS } from '../../url_state/constants';
+
+export const replaceKqlQueryLocationForHostPage = (kqlQuery: string): string => {
```

Comment:
Really nice utility for this redirect workflow (and definitely a gotcha if these props aren't updated!) . If we find ourselves doing this with other integrations we may want to combine this with `replaceKqlQueryLocationForNetworkPage` into a generic solution for updating `queryLocation` / `type`. 

Nothing to change here, just pointing out the above. I did thorough testing of both the network/host page <---> details redirects, and this is all looking 👍